### PR TITLE
set up react environment

### PR DESCRIPTION
**Pull request by rianp**
## Changes made:

1. Set up React environment
2.  Add linter
3. Add continuous integration
4. Add git hook for local linting upon push

## Acceptance criteria:

1. The README should properly document the code and include instructions for running the game.
2. A CI pipeline should be set up on GitHub to automatically run tests, linters and build my application whenever changes are pushed to the repository.
3. Git hooks should run tests on my local machine before pushing to remote repo.
4. Environment version control should be set up for Javascript.
5. Add rule to your repo: PR needs 1 approval before it can be merged (if all comments have been addressed and there is no further response from the reviewers after a day, then you can merge the PR)
6. Setup React testing library for use with tests

[ticket](https://trello.com/c/WClTKpQ8)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
+
+# dependencies
+/node_modules
+/.pnp
+.pnp.js
+
```
Comment (ranizilpelwar) [inline .gitignore:7]: I like that you consider a gitignore from the get-go and included node_modules.

File: README.md [hunk 2]
```diff
@@ -1 +1,30 @@
-# functional-ttt
\ No newline at end of file
+# Tic-Tac-Toe
+A functionally written tic-tac-toe game using React
```
Comment (ranizilpelwar) [inline README.md:2]: I like that you considered the readme and added info about this project, even if its simple at the moment. The details matter. Well done :)

File: package.json [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
+    "react-scripts": "5.0.1",
+    "web-vitals": "^2.1.4"
+  },
+  "scripts": {
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "test": "react-scripts test",
+    "eject": "react-scripts eject"
```
Comment (ranizilpelwar) [inline package.json:18]: I forget what eject does; not sure if that gets included in normal react apps; would be good to hear what the other mentors do.

**Review by ranizilpelwar — APPROVED** [review]

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs
+
+name: Node.js CI
```
Comment (clairenguyen) [inline .github/workflows/ci.yaml]: Just a note for later -- probably should rename this. I'm guessing this template was originally for a backend Node.js app.

File: package.json (hunk 3 above)
Comment (clairenguyen) [reply package.json:18]: So, create-react-app helps bootstrap a bunch of configurations.  You'd `eject` if you no longer want use the defaults and don't want create-react-app to manage the dependencies/configurations for you. 

From [FreeCodeCamp](https://www.freecodecamp.org/news/create-react-app-npm-scripts-explained/): 
> The last command, eject, is used to remove the dependency on react-scripts and expose the build tools and configurations for you to modify.

> All the configuration files from react-scripts will be copied into your project root's config/ folder, and the scripts to run the build will be copied into the scripts/ folder. The dependencies will also be moved into your root's package.json file.

> This command is a one-way operation. Once you have ejected from CRA set up, you can't undo it.

I personally wouldn't mind leaving it in there so you know it's an option that is specific to create-react-app. It'd get deleted if you end up actually ejecting it in the future in order to use your own configs.

File: src/setupTests.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+// jest-dom adds custom jest matchers for asserting on DOM nodes.
+// allows you to do things like:
+// expect(element).toHaveTextContent(/react/i)
+// learn more: https://github.com/testing-library/jest-dom
+import '@testing-library/jest-dom';
```
Comment (clairenguyen) [inline src/setupTests.js:5]: Not a requirement but you may want to consider adding the `jest-extended` library for some cool additional matchers.  

https://jest-extended.jestcommunity.dev/docs/matchers

**Review by clairenguyen — APPROVED** [review]
Added some nice-to-have comments. Good job :)

File: package.json (hunk 3 above)
Comment (ranizilpelwar) [reply package.json:18]: thanks Claire!

File: src/setupTests.js (hunk 5 above)
Comment (rianp) [reply src/setupTests.js:5]: awesome this looks great, I'll add it!

File: package.json (hunk 3 above)
Comment (rianp) [reply package.json:18]: Thanks! I'll keep it in the ReadMe then!

**Pull request merged**